File: src/main/java/com/meetify/server/model/Id.java
package com.meetify.server.model;

import javax.persistence.Embeddable;
import java.io.Serializable;

/**
 * Created by dmitry on 10/14/16.
 */

@Embeddable
public class Id extends JsonParser implements Serializable {

    private Long id;

    public Id() {
    }

    public Id(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Id id1 = (Id) o;

        return id != null ? id.equals(id1.id) : id1.id == null;

    }

    @Override
    public int hashCode() {
        return id != null ? id.hashCode() : 0;
    }
}

File: src/main/java/com/meetify/server/model/Id.kt
package com.meetify.server.model

import java.io.Serializable
import javax.persistence.Column
import javax.persistence.Embeddable

/**
 * This class is embedded id for all entity classes.
 * It's expressed in [BaseEntity] which contains instance of Id.
 * @version     0.0.1
 * @since       0.0.1
 * @property    id  id
 * @constructor takes [id].
 */
@Embeddable
data class Id(@Column(insertable = false, updatable = false)
              var id: Long = 0) : Serializable
